```

[2026-02-15T13:39:26.832Z] ======================================================================

[2026-02-15T13:39:26.832Z] FAIL: test_value_00 (tests.data.test_box_utils.TestCreateBoxList.test_value_00)

[2026-02-15T13:39:26.832Z] ----------------------------------------------------------------------

[2026-02-15T13:39:26.832Z] Traceback (most recent call last):

[2026-02-15T13:39:26.832Z]   File "/usr/local/lib/python3.12/dist-packages/parameterized/parameterized.py", line 620, in standalone_func

[2026-02-15T13:39:26.832Z]     return func(*(a + p.args), **p.kwargs, **kw)

[2026-02-15T13:39:26.832Z]            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

[2026-02-15T13:39:26.832Z]   File "/home/jenkins/agent/workspace/MONAI-premerge/monai/tests/data/test_box_utils.py", line 214, in test_value

[2026-02-15T13:39:26.832Z]     assert_allclose(nms_box, [1, 2, 0], type_test=False)

[2026-02-15T13:39:26.832Z]   File "/home/jenkins/agent/workspace/MONAI-premerge/monai/tests/test_utils.py", line 163, in assert_allclose

[2026-02-15T13:39:26.832Z]     np.testing.assert_allclose(actual, desired, *args, **kwargs)

[2026-02-15T13:39:26.832Z]   File "/usr/local/lib/python3.12/dist-packages/numpy/testing/_private/utils.py", line 1504, in assert_allclose

[2026-02-15T13:39:26.832Z]     assert_array_compare(compare, actual, desired, err_msg=str(err_msg),

[2026-02-15T13:39:26.832Z]   File "/usr/lib/python3.12/contextlib.py", line 81, in inner

[2026-02-15T13:39:26.832Z]     return func(*args, **kwds)

[2026-02-15T13:39:26.832Z]            ^^^^^^^^^^^^^^^^^^^

[2026-02-15T13:39:26.832Z]   File "/usr/local/lib/python3.12/dist-packages/numpy/testing/_private/utils.py", line 797, in assert_array_compare

[2026-02-15T13:39:26.832Z]     raise AssertionError(msg)

[2026-02-15T13:39:26.832Z] AssertionError: 

[2026-02-15T13:39:26.832Z] Not equal to tolerance rtol=1e-07, atol=0

[2026-02-15T13:39:26.832Z] 

[2026-02-15T13:39:26.832Z] Mismatched elements: 2 / 3 (66.7%)

[2026-02-15T13:39:26.832Z] Max absolute difference: 1

[2026-02-15T13:39:26.832Z] Max relative difference: 1.

[2026-02-15T13:39:26.832Z]  x: array([2, 1, 0])

[2026-02-15T13:39:26.832Z]  y: array([1, 2, 0])

[2026-02-15T13:39:26.832Z] 
```

--- a/monai/data/box_utils.py
+++ b/monai/data/box_utils.py
@@ -1128,7 +1128,8 @@ def non_max_suppression(
     scores_t, *_ = convert_to_dst_type(scores, boxes_t)
 
     # sort boxes in descending order according to the scores
-    sort_idxs = torch.argsort(scores_t, dim=0, descending=True)
+    # use stable=True to ensure deterministic ordering when scores are equal
+    sort_idxs = torch.argsort(scores_t, dim=0, descending=True, stable=True)
     boxes_sort = deepcopy(boxes_t)[sort_idxs, :]
 
     # initialize the list of picked indexes

--- a/runtests.sh
+++ b/runtests.sh
@@ -136,7 +136,7 @@ function print_version {
 
 function install_deps {
     echo "Pip installing MONAI development dependencies and compile MONAI cpp extensions..."
-    ${cmdPrefix}"${PY_EXE}" -m pip install -r requirements-dev.txt
+    ${cmdPrefix}"${PY_EXE}" -m pip install --no-build-isolation -r requirements-dev.txt
 }
 
 function compile_cpp {

